pkg/deployment/autoscale.go

```diff
@@ -46,19 +50,21 @@ func autoscalers(
 		maxReplicas = defaultMaxReplicas
 	}
 
-	componentLabels["app.kubernetes.io/component"] = label
+	componentLabels := component.labels()
+	componentLabels["app.kubernetes.io/component"] = hpaLabel
```

Comment:
Wondering whether instead of using `component.labels()`, could introduce another function (e.g. `component.hpaLabels()`) which includes the specific `app.k8s.io/component` labels - rather than passing it in?

Reply:
it makes sense.